node/src/blockchain/blockchain_bridge.rs

```diff
@@ -653,13 +455,171 @@ mod tests {
+        let mut sent_payments_cloned = sent_payment_msg.payments.clone();
+        sent_payments_cloned
+            .iter_mut()
+            .for_each(|payment_in_result| {
+                if let Ok(payment) = payment_in_result {
+                    payment.timestamp = from_time_t(0)
+                }
+            });
+        assert_eq!(
+            sent_payments_cloned,
+            vec![
+                Ok(Payment {
+                    to: make_wallet("blah"),
+                    amount: 420,
+                    timestamp: from_time_t(0), //cannot be asserted directly, this field is just a sentinel
+                    transaction: H256::from("sometransactionhash".keccak256())
+                }),
+                Ok(Payment {
+                    to: make_wallet("foo"),
+                    amount: 210,
+                    timestamp: from_time_t(0), //cannot be asserted directly, this field is just a sentinel
+                    transaction: H256::from("someothertransactionhash".keccak256())
+                })
+            ]
+        )
+    }
+
+    #[test]
+    fn retrieve_transactions_sends_transactions_to_blockchain_interface() {
+        let system =
+            System::new("retrieve_transactions_sends_transactions_to_blockchain_interface");
+        let (accountant, _, accountant_recording_arc) = make_recorder();
+        let earning_wallet = make_wallet("somewallet");
+        let amount = 42u64;
+        let amount2 = 55u64;
+        let expected_gas_price = 5u64;
+        let expected_transactions = vec![
+            Transaction {
+                block_number: 7u64,
+                from: earning_wallet.clone(),
+                gwei_amount: amount,
+            },
+            Transaction {
+                block_number: 9u64,
+                from: earning_wallet.clone(),
+                gwei_amount: amount2,
+            },
+        ];
+        let blockchain_interface_mock = BlockchainInterfaceMock::default()
+            .retrieve_transactions_result(Ok(expected_transactions.clone()));
+        let persistent_configuration_mock =
+            PersistentConfigurationMock::default().gas_price_result(Ok(expected_gas_price));
+        let subject = BlockchainBridge::new(
+            Box::new(blockchain_interface_mock),
+            Box::new(persistent_configuration_mock),
+            false,
+            Some(earning_wallet.clone()),
+        );
+        let addr = subject.start();
+        let subject_subs = BlockchainBridge::make_subs_from(&addr);
+        let peer_actors = peer_actors_builder().accountant(accountant).build();
+        send_bind_message!(subject_subs, peer_actors);
+
+        let retrieve_transactions = RetrieveTransactions {
+            start_block: 0u64,
+            recipient: earning_wallet.clone(),
+        };
+        addr.try_send(retrieve_transactions)
+            .expect("Blockchain bridge is dead");
```

Comment:
You don't need to use this expect in a test we do that in the production code but unwrap() is just enough here.

You should also keep some alignment which means that above retrieve_transactions wouldn't be a blank line but would be after the block of code so that addr.try_send() stands alone.